stat.go

```diff
@@ -125,6 +125,32 @@ func ChiSquare(obs, exp []float64) float64 {
 	return result
 }
 
+// CircularMean returns the circular mean of the dataset.
+// If weights is nil then all of the weights are 1. If weights is not nil, then
+// len(x) must equal len(weights).
+func CircularMean(x, weights []float64) float64 {
+	if weights != nil && len(x) != len(weights) {
+		panic("stat: slice length mismatch")
+	}
+
+	var aX, aY float64
```

Comment:
Change this as @jonlawlor  suggested, and may as well move the weights check out of the loop

````
var aX, aY float64
if weights != nil {
    for _, v := range x {
        ax += math.Cos(v)
        ay += math.Sin(v)
    }
}else{
    for i, v := range x {
        aX += weights[i]  * math.Cos(v)
        aY += weights[i] * math.Sin(v)
   }
}
````